File: Python/underautomation/universal_robots/primary_interface/joint_configuration.py
import typing
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import JointConfiguration as joint_configuration

class JointConfiguration:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = joint_configuration()
		else:
			self._instance = _internal
	@property
	def joint_min_limit(self) -> float:
		return self._instance.JointMinLimit
	@joint_min_limit.setter
	def joint_min_limit(self, value: float):
		self._instance.JointMinLimit = value
	@property
	def joint_max_limit(self) -> float:
		return self._instance.JointMaxLimit
	@joint_max_limit.setter
	def joint_max_limit(self, value: float):
		self._instance.JointMaxLimit = value
	@property
	def joint_max_speed(self) -> float:
		return self._instance.JointMaxSpeed
	@joint_max_speed.setter
	def joint_max_speed(self, value: float):
		self._instance.JointMaxSpeed = value
	@property
	def joint_max_acceleration(self) -> float:
		return self._instance.JointMaxAcceleration
	@joint_max_acceleration.setter
	def joint_max_acceleration(self, value: float):
		self._instance.JointMaxAcceleration = value
	@property
	def d_ha(self) -> float:
		return self._instance.DHa
	@d_ha.setter
	def d_ha(self, value: float):
		self._instance.DHa = value
	@property
	def d_hd(self) -> float:
		return self._instance.DHd
	@d_hd.setter
	def d_hd(self, value: float):
		self._instance.DHd = value
	@property
	def d_halpha(self) -> float:
		return self._instance.DHalpha
	@d_halpha.setter
	def d_halpha(self, value: float):
		self._instance.DHalpha = value
	@property
	def d_htheta(self) -> float:
		return self._instance.DHtheta
	@d_htheta.setter
	def d_htheta(self, value: float):
		self._instance.DHtheta = value

File: PrimaryInterface/JointConfiguration.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.


namespace PrimaryInterface {
	/// <summary>
	/// Joint configuration
	/// </summary>
	public class JointConfiguration {

		/// <summary>
		/// Minimum angular position in rad
		/// </summary>
		[PackageDescription("Minimum angular position in rad", PackageUnit.Radian)]
public double JointMinLimit { get; set; }

		/// <summary>
		/// Maximum angular position in rad
		/// </summary>
		[PackageDescription("Maximum angular position in rad", PackageUnit.Radian)]
public double JointMaxLimit { get; set; }

		/// <summary>
		/// Maximum rotation speed in rad/s
		/// </summary>
		[PackageDescription("Maximum rotation speed in rad/s", PackageUnit.RadianPerSecond)]
public double JointMaxSpeed { get; set; }

		/// <summary>
		/// Maximum rotation speed in rad/s²
		/// </summary>
		[PackageDescription("Maximum rotation speed in rad/s²", PackageUnit.RadianPerSecondSquared)]
public double JointMaxAcceleration { get; set; }

		/// <summary>
		/// a parameter of Denavit–Hartenberg (DH) convention
		/// </summary>
		[PackageDescription("a parameter of Denavit–Hartenberg (DH) convention", PackageUnit.Meter)]
public double DHa { get; set; }

		/// <summary>
		/// d parameter of Denavit–Hartenberg (DH) convention
		/// </summary>
		[PackageDescription("d parameter of Denavit–Hartenberg (DH) convention", PackageUnit.Meter)]
public double DHd { get; set; }

		/// <summary>
		/// Alpha parameter of Denavit–Hartenberg (DH) convention
		/// </summary>
		[PackageDescription("Alpha parameter of Denavit–Hartenberg (DH) convention", PackageUnit.Radian)]
public double DHalpha { get; set; }

		/// <summary>
		/// Theta parameter of Denavit–Hartenberg (DH) convention
		/// </summary>
		[PackageDescription("Theta parameter of Denavit–Hartenberg (DH) convention", PackageUnit.Radian)]
public double DHtheta { get; set; }
	}
}
